src/formatDate.js

```diff
@@ -50,7 +50,26 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat.pop();
+  const dateParts = date.split(separator);
+  const dateMap = {};
+
+  fromFormat.forEach((format, index) => {
+    dateMap[format] = dateParts[index];
+  });
+
+  if (dateMap['YYYY'] && toFormat.includes('YY')) {
+    dateMap['YY'] = dateMap['YYYY'].slice(-2);
+  } else if (dateMap['YY'] && toFormat.includes('YYYY')) {
+    const year = parseInt(dateMap['YY'], 10);
+
+    dateMap['YYYY'] = year < 30 ? `20${dateMap['YY']}` : `19${dateMap['YY']}`;
+  }
+
+  const newSeparator = toFormat.pop();
```

Comment:
Similar to the previous comment on line 53, using `pop` on `toFormat` mutates the original array. It's better to reference the last element without removing it.